src/modules/todos/controllers.js

```diff
@@ -1,144 +1,154 @@
-import { todos} from '../database/database.js';
-import AppException from '../../utils/exceptions/AppException.js';
-import { v4 } from 'uuid';
+import { v4 } from "uuid";
 
+import pool from "../../database/database.js";
+import AppException from "../../utils/exceptions/AppException.js";
+import queries from "./queries.js";
 
-export function getTodos(req, res) {
-	res.json(todos.filter((todo) => todo.userId === req.userId));
-};
+export async function getTodos(req, res, next) {
+	try {
+		const allTodos = await pool.query(queries.allTodos, [req.userId]);
+		res.json(allTodos.rows);
+	} catch (err) {
+		next(err);
+	}
+}
 
-export function createTodo(req, res, next) {
+export async function createTodo(req, res, next) {
 	try {
 		let today = new Date();
-		
-		let strDate = 'Y-m-d h:M:s'
-			.replace('Y', today.getFullYear())
-			.replace('m', today.getMonth()+1)
-			.replace('d', today.getDate())
-			.replace('h', today.getHours())
-			.replace('M', today.getMinutes())
-			.replace('s', today.getSeconds());
-		const id = v4();
-		const userId = req.userId;
-		const title = req.body.title;
-		const status = req.body.status;
-		const dueDate = req.body.dueDate;
+
+		let strDate = "Y-m-d h:M:s"
+			.replace("Y", today.getFullYear())
+			.replace("m", today.getMonth() + 1)
+			.replace("d", today.getDate())
+			.replace("h", today.getHours())
+			.replace("M", today.getMinutes())
+			.replace("s", today.getSeconds());
+		const todo_id = v4();
+		const user_id = req.userId;
+		const { title, status, dueDate } = req.body;
 		const createdAt = strDate;
 		const updatedAt = strDate;
 		const todo = {
-			id,
-			userId,
+			todo_id,
+			user_id,
 			title,
 			status,
 			dueDate,
 			createdAt,
 			updatedAt,
-		}
-		todos.push(todo);
+		};
+		const newTodo = await pool.query(queries.newTodo, [
+			todo.todo_id,
+			todo.user_id,
+			todo.title,
+			todo.status,
+			todo.dueDate,
+			todo.createdAt,
+			todo.updatedAt,
+		]);
 		return res.status(201).json({
-			statcode: 201,
-			data: {
-				id: todo.id,
-				userId: todo.userId,
-				title: todo.title,
-				status: todo.status,
-				dueDate: todo.dueDate,
-				createdAt: todo.createdAt,
-				updatedAt: todo.updatedAt,
-			}
-		})
+			statuscode: 201,
+			data: newTodo.rows[0],
```

Comment:
Avoid returning the entire response from the DB. Why: Imagine if down the line, another developer adds some sensitive column to that table and does not know you wrote this like this. You would be exposing that sensitive data.